src/commands/shell.mjs

```diff
@@ -1,23 +1,37 @@
 //@ts-check
 
+import path from "node:path";
 import repl from "node:repl";
 
 import { container } from "../cli.mjs";
 import {
   validateDatabaseOrSecret,
   yargsWithCommonConfigurableQueryOptions,
 } from "../lib/command-helpers.mjs";
-import { getSecret } from "../lib/fauna-client.mjs";
+import { formatQueryResponse, getSecret } from "../lib/fauna-client.mjs";
+import { dirExists, fileExists } from "../lib/file-util.mjs";
 
 async function shellCommand(argv) {
   validateDatabaseOrSecret(argv);
 
+  const fs = container.resolve("fs");
   const logger = container.resolve("logger");
   let completionPromise;
 
   if (argv.dbPath) logger.stdout(`Starting shell for database ${argv.dbPath}`);
   logger.stdout("Type Ctrl+D or .exit to exit the shell");
 
+  // Setup history file
+  const homedir = container.resolve("homedir")();
+  const historyDir = path.join(homedir, ".fauna");
+  if (!dirExists(historyDir)) {
+    fs.mkdirSync(historyDir, { recursive: true });
+  }
+  const historyFile = path.join(historyDir, "history");
+  if (!fileExists(historyFile)) {
+    fs.writeFileSync(historyFile, "");
+  }
+
```

Comment:
put the file management code in `src/lib/file-util.mjs` which is where the other files we manage are maintained.

Reply:
doesn't credential setup happen in there?

Reply:
doh... CredentialStorage is indeed in there. I was trying to go off of memory and I was wrong.

Reply:
I added new functions to file-util to initialize and clear history storage.